tf_fastmri_data/datasets/non_cartesian.py

```diff
@@ -82,11 +93,16 @@ def preprocessing(self, image, kspace):
             )
         traj = tf.repeat(traj, tf.shape(image)[0], axis=0)
         orig_image_channels = ortho_ifft2d(kspace)
-        nc_kspace = nufft(self.nufft_obj, orig_image_channels, traj, self.image_size)
+        nc_kspace = nufft(self.nufft_obj, orig_image_channels, traj, self.image_size, multicoil=self.multicoil)
         nc_kspace, image = scale_tensors(nc_kspace, image, scale_factor=self.scale_factor)
         image = image[..., None]
         nc_kspaces_channeled = nc_kspace[..., None]
-        orig_shape = tf.ones([tf.shape(kspace)[0]], dtype=tf.int32) * tf.shape(kspace)[-1]
+        orig_shape = tf.ones([tf.shape(kspace)[0]], dtype=tf.int32) * tf.shape(self.image_size)[-1]
+        if self.crop_image_data:
+            image = adjust_image_size(image, self.image_size)
```

Comment:
I am wondering if in some cases there are brain images which are less (in some dimension) than `320 x 320`. If so, you will have an output dimension mismatch if you don't specify the output shape, even if you did adjust the image size.

Could you check that?

Reply:
Wait no, if the dimension is low, cant we still pad the image? Is this like a PR to `fastmri_recon`? 
My point is that we should use `adjust_image_size` wherever possible. That way, we will crop or pad as needed and not have any issues.. What do you think?

Reply:
I am slightly less comfortable using padding at the end of the network to fit the expected output image size.
Indeed it brings the question of how exactly to pad because you don't exactly have a 0 background.

It probably isn't a good reason from an MRI point of view but I feel less easy about it.

But even so, you would still need to pass in the output image shape to do padding, so it's not a question for this PR.

Reply:
Yes but it does become a question whether i should throw error or warn in case the data shape is less than 320

Reply:
rather less than the expected output shape in the cross domain net

Reply:
Yes... Which is `image_size` . I will add a check and add a warning for now..